casper/src/test/resources/PoSTest.rho

```diff
@@ -4,31 +4,248 @@
 
 match (
   "cc87bf7747a8c176714b417ca14a63897b07572876c5e38a7896b6007738ef81".hexToBytes(),
-  "11112GofKy1v5mPWZjZEYBByvpZ4ahYJp6vFN1hgGCn7L543rFSpyA"
+  "1111111111111111111111111111111111111111111111111111111111111111".hexToBytes(),
+  "2222222222222222222222222222222222222222222222222222222222222222".hexToBytes()
 ) {
 
-  (genesisPubKey, genesisRevAddress) => {
+  (genesisPubKey, user1PubKey, user2PubKey) => {
+    new
+      rl(`rho:registry:lookup`),
+      stdlog(`rho:io:stdlog`),
+      stdout(`rho:io:stdout`),
+      getParameters(`rho:deploy:params`),
+      setDeployData(`rho:test:deploy:set`),
+      revAddressOps(`rho:rev:address`),
+      posRevAddressCh, posVaultCh, user1RevAddressCh, user1VaultCh, user2RevAddressCh, user2VaultCh,
+      RhoSpecCh, MakePoSCh, RevVaultCh,
+      setup,
+      test_make_pos_succeeds, test_bonding_succeeds,
+      test_multiple_bonding_succeeds,
+      test_bonding_fails_if_deposit_fails,
+      test_bonding_fails_if_already_bonded,
+      test_bonding_fails_if_bond_too_small,
+      test_bonding_fails_if_bond_too_big,
+      test_pay_succeeds
+    in {
+      revAddressOps!("fromPublicKey", genesisPubKey, *posRevAddressCh) |
+      revAddressOps!("fromPublicKey", user1PubKey, *user1RevAddressCh) |
+      revAddressOps!("fromPublicKey", user2PubKey, *user2RevAddressCh) |
+      rl!(`rho:id:1o93uitkrjfubh43jt19owanuezhntag5wh74c6ur5feuotpi73q8z`, *RevVaultCh) |
+      rl!(`rho:id:cnec3pa8prp4out3yc8facon6grm3xbsotpd4ckjfx8ghuw77xadzt`, *MakePoSCh) |
+      rl!(`rho:id:6wnujzcraztjfg941skrtbdkdgbko8nuaqihuhn15s66oz8ro5gwbb`, *RhoSpecCh) |
+      for(@(_, RhoSpec) <- RhoSpecCh;
+          @(_, MakePoS) <- MakePoSCh;
+          @(_, RevVault) <- RevVaultCh;
+          @posRevAddress <- posRevAddressCh;
+          @user1RevAddress <- user1RevAddressCh;
+          @user2RevAddress <- user2RevAddressCh) {
+        stdlog!("info", {"posRevAddress":posRevAddress, "user1RevAddress":user1RevAddress}) |
 
-    new rl(`rho:registry:lookup`),
-        RhoSpecCh,
-        PosCh,
-        setup,
-        test,
-        test_different_deployer,
-        ackCh in {
+        @RevVault!("findOrCreateGenesisVault", posRevAddress, 10000, *posVaultCh) |
```

Comment:
I don't think it's necessary to instantiate the `PoS` vault here. REV won't be leaving the `PoS` vault, and it's instantiated in the `PoS` constructor.

Reply:
It is created. It's created in line 43 of `PoS.rho`.